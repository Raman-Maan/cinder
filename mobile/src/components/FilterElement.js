import React from 'react';
import { StyleSheet, Image } from 'react-native';
import { H1, View, Text, Container, Button } from 'native-base';

import FilterModal from './FilterModal';

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
  },
  image: {
    height: 100,
    width: 100,
    marginTop: 20,
    borderRadius: 50,
  },
  filterContainer: {
    marginTop: 40,
  },
  userName: {
    marginTop: 10,
  },
  lineStyle: {
    alignSelf: 'stretch',
    borderWidth: 0.5,
    borderColor: 'grey',
    marginTop: 15,
    marginRight: 2,
    marginLeft: 2,
  },
});

export default class FilterElement extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    return (
      <View style={styles.container}>
        <Button onPress={this.props.onBackdropPress} />
        <FilterModal
          {...this.props}
        />
      </View>
    );
  }
}
