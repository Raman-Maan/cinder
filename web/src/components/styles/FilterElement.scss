@import '../../assets/bootstrap/bootstrap.scss';

$height: 30px;
$width: 52px;
$spacing: 4px;
$transition: 300ms ease-in-out;
$bg-color: $gray-300;
$color: $white;

.switch {
  position: relative;
  height: $height;
  width: $width;
  margin: 0;

  input {
    display: none;
  }

  .slider {
    background-color: $bg-color;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;

    cursor: pointer;
    transition: $transition;

    &.round {
      border-radius: 34px;

      &::before {
        border-radius: 50%;
      }
    }

    &::before {
      content: '';
      height: $height - 2*$spacing;
      width: $height - 2*$spacing;
      background-color: $color;

      position: absolute;
      bottom: $spacing;
      left: $spacing;
      transition: $transition;
    }
  }

  input:checked + .slider {
    background-color: $primary;
    &::before {
      transform: translateX(100%);
    }
  }
}

$filter-spacing: 4px;
.filter.element {
  display: flex;
  width: 100%;
  height: 100%;
  justify-content: center;
  align-content: center;
  flex-direction: row;
  border: 1px solid $bg-color;
  border-radius: $spacing;
  padding: $spacing;

  .toggle {
    display: flex;
    margin: $filter-spacing;
    align-items: center;
  }

  .content {
    position: relative;
    margin: $filter-spacing;
    align-content: center;
    min-height: 100%;
    width: 100%;
  }

  &.disabled {
    background-color: lighten($bg-color, 10%);
  }
}
