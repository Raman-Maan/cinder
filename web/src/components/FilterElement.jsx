import React from 'react';

import './styles/FilterElement.css';

export default class FilterElement extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      disabled: !this.props.checked
    };
  }

  toggle() {
    this.props.onChange(this.state.disabled);
    this.setState(prevState => {
      return { disabled: !prevState.disabled };
    });
  }

  renderSwitch() {
    return (
      <label className="switch">
        <input
          type="checkbox"
          checked={this.props.checked}
          onChange={() => this.toggle()}
        />
        <div className={`slider ${this.props.round ? 'round' : ''}`} />
      </label>
    );
  }

  render() {
    return (
      <div
        className={`filter element ${this.state.disabled ? 'disabled' : ''}`}
      >
        <div className="toggle">{this.renderSwitch()}</div>
        <div className="content">{this.props.children}</div>
      </div>
    );
  }
}
