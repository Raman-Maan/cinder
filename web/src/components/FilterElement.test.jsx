import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import FilterElement from './FilterElement';

describe('<FilterElement />', () => {
  describe('Snapshot tests', () => {
    it('should render with no children - disabled by default', () => {
      const component = renderer.create(<FilterElement />);
      expect(component).toMatchSnapshot();
    });

    it('should render with children', () => {
      const component = renderer.create(
        <FilterElement>
          <span>Some content here</span>
        </FilterElement>
      );
      expect(component).toMatchSnapshot();
    });

    it('should render round if round', () => {
      const component = renderer.create(<FilterElement round />);
      expect(component).toMatchSnapshot();
    });

    it('should render not disabled if checked', () => {
      const component = renderer.create(<FilterElement checked />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Enzyme tests', () => {
    let wrapper;
    let mockActiveChange;

    beforeEach(() => {
      mockActiveChange = jest.fn();
      wrapper = mount(<FilterElement round onChange={mockActiveChange} />);
    });

    it('should toggle', () => {
      expect(wrapper.find('.filter.element.disabled')).toHaveLength(1);
      expect(wrapper.state('disabled')).toEqual(true);

      expect(mockActiveChange.mock.calls).toHaveLength(0);
      wrapper.find('input').simulate('change');
      expect(wrapper.find('filter.element.disabled')).toHaveLength(0);
      expect(wrapper.state('disabled')).toEqual(false);

      expect(mockActiveChange.mock.calls).toHaveLength(1);
    });

    it('should trigger onChange on toggle', () => {
      expect(mockActiveChange.mock.calls).toHaveLength(0);
      wrapper.find('input').simulate('change');

      expect(mockActiveChange.mock.calls).toHaveLength(1);
      wrapper.find('input').simulate('change');

      expect(mockActiveChange.mock.calls).toHaveLength(2);
      wrapper.find('input').simulate('change');

      expect(mockActiveChange.mock.calls).toHaveLength(3);
      wrapper.find('input').simulate('change');
    });
  });
});
