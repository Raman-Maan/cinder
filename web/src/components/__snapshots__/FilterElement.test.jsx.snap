// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterElement /> Snapshot tests should render not disabled if checked 1`] = `
<div
  className="filter element"
>
  <div
    className="toggle"
  >
    <label
      className="switch"
    >
      <input
        checked={true}
        onChange={undefined}
        type="checkbox"
      />
      <div
        className="slider "
      />
    </label>
  </div>
  <div
    className="content"
  />
</div>
`;

exports[`<FilterElement /> Snapshot tests should render round if round 1`] = `
<div
  className="filter element"
>
  <div
    className="toggle"
  >
    <label
      className="switch"
    >
      <input
        checked={undefined}
        onChange={undefined}
        type="checkbox"
      />
      <div
        className="slider round"
      />
    </label>
  </div>
  <div
    className="content"
  />
</div>
`;

exports[`<FilterElement /> Snapshot tests should render with children 1`] = `
<div
  className="filter element"
>
  <div
    className="toggle"
  >
    <label
      className="switch"
    >
      <input
        checked={undefined}
        onChange={undefined}
        type="checkbox"
      />
      <div
        className="slider "
      />
    </label>
  </div>
  <div
    className="content"
  >
    <span>
      Some content here
    </span>
  </div>
</div>
`;

exports[`<FilterElement /> Snapshot tests should render with no children - disabled by default 1`] = `
<div
  className="filter element"
>
  <div
    className="toggle"
  >
    <label
      className="switch"
    >
      <input
        checked={undefined}
        onChange={undefined}
        type="checkbox"
      />
      <div
        className="slider "
      />
    </label>
  </div>
  <div
    className="content"
  />
</div>
`;
